tests/integrationtest/t/ddl/constraint.test

```diff
@@ -658,4 +658,23 @@ show create table t;
 # _, err = tk.Exec("alter table t alter constraint haha enforced")
 # require.Errorf(t, err, "[table:3819]Check constraint 'haha' is violated.")
 
-set @@global.tidb_enable_check_constraint = 0;
\ No newline at end of file
+# Related issue TiDB#47567, #47631 and #47632.
+# TestCheckConstraintIssue47567
+drop table if exists t;
+set @@global.tidb_enable_check_constraint = 1;
+CREATE TABLE `t` (`a` int(11) DEFAULT NULL);
+show create table t;
+insert t values(1);
+select * from t;
+-- error 3940
+alter table t ADD CONSTRAINT chk CHECK (a > 1) ENFORCED;
+-- error 3940
+alter table t ADD CONSTRAINT chk CHECK (a > 1) ENFORCED;
```

Comment:
duplicated with line 669-670?

Reply:
No, it is desired. https://github.com/pingcap/tidb/pull/47633#discussion_r1375588267 
Same as it.

Reply:
> issue #47632 reproduce steps，the first sql will rollback failed, and the second will report "Duplicate check constraint name" error.

Get it